opba-protocols/xs2a-protocol/src/main/java/de/adorsys/opba/protocol/xs2a/entrypoint/authorization/Xs2aUpdateAuthorization.java

```diff
@@ -83,6 +84,10 @@ private void updateWithScaChallenges(Xs2aContext context, Map<String, String> sc
 
         if (null != scaChallenges.get(SCA_CHALLENGE_ID)) {
             context.setUserSelectScaId(scaChallenges.get(SCA_CHALLENGE_ID));
+            context.setUserSelectScaType(context.getAvailableSca().stream()
+                    .filter(it -> it.getKey().equals(context.getUserSelectScaId()))
```

Comment:
Please invert condition as `context.getUserSelectScaId()` can't be null, but we can't do same statement on `it.getKey()`. Improbable, but we are 100% sure that `context.getUserSelectScaId()` is not null